### PR TITLE
feat: merge snippet props with existing props from yaml

### PR DESCRIPTION
**Pull request by p-spacek**
### What does this PR do?
merge snippet props with existing props from the yaml

#### example
existing yaml:

```yaml
object:
  prop1: some value
  # invoke completion here
```

snippet:
```
defaultSnippets:[
  {
    label: "example1",
    body: {
      prop1: {},
      prop2: {} 
   }
]
```

by this PR, it will include only `prop2` into the inserted text, because `prop1` is already in the yaml

note that only the first level of the props is checked

### What issues does this PR fix or reference?
no ref

### Is it tested? How?
added tests and modified existings

### CODE REVIEW

File: src/languageservice/services/yamlCompletion.ts [hunk 1]
```diff
@@ -286,6 +288,7 @@ export class YamlCompletion {
         return result.items.length;
       },
       result,
+      proposed,
```
Comment (p-spacek) [inline src/languageservice/services/yamlCompletion.ts:291]: to be able to access it by `collector`

File: src/languageservice/utils/json.ts [hunk 2]
```diff
@@ -52,24 +53,32 @@ export function stringifyObject(
         return '';
       }
       let result = (depth === 0 && settings.newLineFirst) || depth > 0 ? '\n' : '';
+      let isFirstProp = true;
       for (let i = 0; i < keys.length; i++) {
         const key = keys[i];
+
+        if (depth === 0 && settings.existingProps.includes(key)) {
```
Comment (p-spacek) [inline src/languageservice/utils/json.ts:60]: main piece of the code that excludes existing props

File: src/languageservice/utils/json.ts [hunk 3]
```diff
@@ -52,24 +53,32 @@ export function stringifyObject(
...
+        if (depth === 0 && isFirstProp && !settings.indentFirstObject) {
+          const value = stringifyObject(obj[key], objectIndent, stringifyLiteral, settings, depth + 1, 0);
+          result += lineBreak + indent + key + colonDelimiter + value;
         } else {
-          const value = stringifyObject(obj[key], objectIndent, stringifyLiteral, settings, (depth += 1), 0);
-          result += newIndent + key + colonDelimiter + value;
-        }
-        if (i < keys.length - 1) {
-          result += '\n';
+          const value = stringifyObject(obj[key], objectIndent, stringifyLiteral, settings, depth + 1, 0);
```
Comment (p-spacek) [inline src/languageservice/utils/json.ts:77]: I believe, that previous code `(depth += 1)` was not correct
it increased depthLevel for each iteration

**Review by gorkem — APPROVED** [review]
LGTM

**Comment by gorkem** [comment]
merged, Thanks for the contribution

**Pull request closed without merging**